rpcs3/rpcs3qt/user_manager_dialog.cpp

```diff
@@ -0,0 +1,320 @@
+		m_table->horizontalHeader()->height() + m_table->verticalHeader()->length() + m_table->frameWidth() * 2);
+
+	QSize preferredSize = minimumSize().expandedTo(sizeHint() - m_table->sizeHint() + tableSize).expandedTo(size());
+	QSize maxSize = QSize(preferredSize.width(), static_cast<int>(QApplication::desktop()->screenGeometry().height()*.6));
+
+	resize(preferredSize.boundedTo(maxSize));
+}
+
+//Remove a user folder, needs to be confirmed.
+void user_manager_dialog::OnUserRemove()
+{
+	int idx = m_table->currentRow();
+	if (idx < 0)
+	{
+		return;
+	}
+	int idx_real = m_table->item(idx, 0)->data(Qt::UserRole).toInt();
+	if (m_user_list[idx_real]->GetUserId() == Emu.GetUsr())
+	{
+		QToolTip::showText(QCursor::pos(), tr("You cannot delete the logged in user!"));
+		return;
+	}
+	if (QMessageBox::question(this, tr("Delete Confirmation"), tr("Are you sure you want to delete:\n%1?").arg(qstr(m_user_list[idx_real]->GetUserName())), QMessageBox::Yes, QMessageBox::No) == QMessageBox::Yes)
+	{
+		fs::remove_all(m_user_list[idx_real]->GetUserDir());
+		user_manager_dialog::UpdateTable();
+	}
+}
+
+void user_manager_dialog::GenerateUser(const std::string& username)
+{
+	// If the user list is sorted by defult from fs::dir, then we just need the last one in the list.
+	std::string largestUserId = m_user_list[m_user_list.size() - 1]->GetUserId();
+
+	// Add one to the largest user id, then reformat the result into an 8-digit string.
+	u8 nextLargest = static_cast<u8>(std::stoul(largestUserId) + 1u);
+	char* buf;
+	sprintf(buf, "%08d", nextLargest);
+	std::string nextUserId(buf);
+
+	// Create user folders and such.
+	const std::string dev_hdd0 = Emu.GetHddDir();
+	fs::create_dir(dev_hdd0 + "home/");
+	fs::create_dir(dev_hdd0 + "home/" + nextUserId + "/");
+	fs::create_dir(dev_hdd0 + "home/" + nextUserId + "/exdata/");
+	fs::create_dir(dev_hdd0 + "home/" + nextUserId + "/savedata/");
+	fs::create_dir(dev_hdd0 + "home/" + nextUserId + "/trophy/");
+	fs::write_file(dev_hdd0 + "home/" + nextUserId + "/localusername", fs::create + fs::excl + fs::write, username);
+}
+
+void user_manager_dialog::OnUserCreate()
+{
+	user_create_dialog* createDialog = new user_create_dialog(m_dir, this);
+	createDialog->setModal(true);
+	connect(createDialog->GetOkButton(), &QAbstractButton::clicked, this, [=]
+	{
+		// "Entire string (^...$) must be between 3 and 16 characters
+		// and only consist of letters, numbers, underscores, and hyphens."
+		QRegExpValidator* validator = new QRegExpValidator(QRegExp("^[A-Za-z0-9_-]{3,16}$"));
+
+		int pos = 0;
+		QString textToValidate = createDialog->GetText();
+		if (validator->validate(textToValidate, pos) != QValidator::Acceptable)
+		{
+			QToolTip::showText(QCursor::pos(), tr("Username must have between 3 and 16 characters \nand only contain letters, numbers, underscores, and dashes."));
+			return;
+		}
+		user_manager_dialog::GenerateUser(textToValidate.toStdString());
```

Comment:
I think writing out the class can be omitted, it takes longer to read and is unnecessary in my opinion